auth/provider_config.go

```diff
@@ -40,6 +119,231 @@ type SAMLProviderConfig struct {
 	CallbackURL           string
 }
 
+// SAMLProviderConfigToCreate represents the options used to create a new SAMLProviderConfig.
+type SAMLProviderConfigToCreate struct {
+	id     string
+	params nestedMap
+}
+
+// ID sets the provider ID of the new config.
+func (config *SAMLProviderConfigToCreate) ID(id string) *SAMLProviderConfigToCreate {
+	config.id = id
+	return config
+}
+
+// IDPEntityID sets the IDPEntityID field of the new config.
+func (config *SAMLProviderConfigToCreate) IDPEntityID(entityID string) *SAMLProviderConfigToCreate {
+	config.set(idpEntityIDKey, entityID)
+	return config
+}
+
+// SSOURL sets the SSOURL field of the new config.
+func (config *SAMLProviderConfigToCreate) SSOURL(url string) *SAMLProviderConfigToCreate {
+	config.set(ssoURLKey, url)
+	return config
```

Comment:
Very minor thing... some places in the code we set the values first and then return the `config` object and in other places we do a single line return `return config.set(ssoURLKey, url)`. It might be better to keep it consistent everywhere

Reply:
Good catch. Fixed.